routers/web/repo/actions/view.go

```diff
@@ -588,7 +642,17 @@ func Approve(ctx *context_module.Context) {
 			return err
 		}
 		for _, job := range jobs {
-			if len(job.Needs) == 0 && job.Status.IsBlocked() {
+			blockJobByConcurrency, err := actions_model.ShouldBlockJobByConcurrency(ctx, job)
+			if err != nil {
+				if actions_model.IsErrUnevaluatedConcurrency(err) {
+					continue
+				}
```

Comment:
Not quite sure, is it a user error (need to tell users that their workflow is wrong)? Or ignore-able error (then a job won't be executed? still need to notify end users) ? Or self-recoverable error (it will be correctly run after the approval)?